tests/unit/swagger/schemaDetails3.spec.ts

```diff
@@ -359,11 +359,20 @@ describe('Definition generation for OpenAPI 3.0.0', () => {
                     expect(propertySchema).to.not.haveOwnProperty('additionalProperties', `for property ${propertyName}`);
                     expect(propertySchema.nullable).to.eq(true, `for property ${propertyName}.nullable`);
                 },
-                genericNested: (propertyName, propertySchema) => {
-                    expect(propertySchema.$ref).to.eq('#/components/schemas/GenericRequestArrayTypeAliasModel1');
+                genericMultiNested: (propertyName, propertySchema) => {
+                    expect(propertySchema.$ref).to.eq('#/components/schemas/GenericRequestGenericRequestTypeAliasModel1', `for property ${propertyName}.$ref`);
                 },
-                genericNested2: (propertyName, propertySchema) => {
-                    expect(propertySchema.$ref).to.eq('#/components/schemas/GenericRequestArrayTypeAliasModel2');
+                genericNestedArrayKeyword1: (propertyName, propertySchema) => {
```

Comment:
@WoH I had to rename `genericNested2` to differentiate between the `[]` tests.  I also added `genericMultiNested` to see if we would have an issue with multiple levels of generics. Turns out that we don't because your function is recursive. Nice work! :)